qiskit/assembler/assemble_schedules.py

```diff
@@ -101,8 +102,9 @@ def _assemble_experiments(
     user_pulselib = {}
     experiments = []
     for idx, schedule in enumerate(schedules):
+        compressed_schedule = compress_pulses(schedule)
```

Comment:
Would it be possible to compress all the pulses across all schedules to go in the qobj? Or would this require a large rewrite? Something like `schedule, updated_pulse_cache = transforms.compress_pulses(schedule, pulse_cache)` or `schedules = transforms.compress_pulses(schedules)`

Reply:
How about this https://github.com/Qiskit/qiskit-terra/pull/4114/commits/75c0523d2d283041a4224069002b1cf523e39faa?